src/vaadin-contextmenu-items-mixin.html

```diff
@@ -0,0 +1,251 @@
+        }
+
+        if (component.localName === 'vaadin-item') {
+          component.setAttribute('role', 'menuitem');
+        } else if (component.localName === 'hr') {
+          component.setAttribute('role', 'separator');
+        }
+
+        component.classList.add('vaadin-menu-item');
+        component._item = item;
+
+        if (item.text) {
+          component.textContent = item.text;
+        }
+
+        this.__toggleMenuComponentAttribute(component, 'menu-item-checked', item.checked);
+        this.__toggleMenuComponentAttribute(component, 'disabled', item.disabled);
+
+        component.setAttribute('aria-haspopup', 'false');
+        component.classList.remove('vaadin-context-menu-parent-item');
+        if (item.children && item.children.length) {
+          component.classList.add('vaadin-context-menu-parent-item');
+          component.setAttribute('aria-haspopup', 'true');
+          component.setAttribute('aria-expanded', 'false');
+        }
+
+        listBox.appendChild(component);
+      });
+    }
+
+    __toggleMenuComponentAttribute(component, attribute, on) {
+      if (on) {
+        component.setAttribute(attribute, '');
+        component['__has-' + attribute] = true;
+      } else if (component['__has-' + attribute]) {
+        component.removeAttribute(attribute);
+        component['__has-' + attribute] = false;
+      }
+    }
+
+    __initMenu(root, menu) {
+      if (!root.firstElementChild) {
+        root.innerHTML = `
+          <vaadin-list-box></vaadin-list-box>
+          <vaadin-context-menu hidden></vaadin-context-menu>
+        `;
+        Polymer.flush();
+        const listBox = root.querySelector('vaadin-list-box');
+        listBox.classList.add('vaadin-menu-list-box');
+        requestAnimationFrame(() => listBox.setAttribute('role', 'menu'));
+        const subMenu = root.querySelector('vaadin-context-menu');
+        subMenu.$.overlay.modeless = true;
+        subMenu.openOn = 'opensubmenu';
+
+        menu.addEventListener('opened-changed', e => !e.detail.value && subMenu.close());
+        subMenu.addEventListener('opened-changed', e => {
+          if (!e.detail.value) {
+            const expandedItem = listBox.querySelector('[aria-expanded="true"]');
+            if (expandedItem) {
+              expandedItem.setAttribute('aria-expanded', 'false');
+            }
+          }
+        });
+
+        listBox.addEventListener('selected-changed', e => {
+          if (typeof e.detail.value === 'number') {
+            const item = e.target.items[e.detail.value]._item;
+            if (!item.children) {
+              const detail = {value: item};
+              menu.dispatchEvent(new CustomEvent('item-selected', {detail}));
+            }
+            listBox.selected = null;
+          }
+        });
+
+        subMenu.addEventListener('item-selected', e => {
+          menu.dispatchEvent(new CustomEvent('item-selected', {detail: e.detail}));
+          menu.close();
```

Comment:
Now the menu closes only when clicking items in a sub-menu.
Clicking a non-parent item in the main overlay doesn't close it.

Reply:
Good catch, done.